src/main/GCNode.cpp

```diff
@@ -387,82 +295,3 @@ void GCNode::restoreInternalData(InternalData data) {
     m_rcmms = data;
 }
 
-#ifdef HAVE_ADDRESS_SANITIZER
-
-/*
- * When compiling with the address sanitizer, we modify the allocation routines
- * to detect errors more reliably.  In particular:
- * - A redzone is added on both sides of the object to detect underflows and
- *   overflows.
- * - Memory is poisoned when it is freed to detect use-after-free errors.
- * - Freed objects are held in a quarantine to prevent the memory from being
- *   reallocated quickly.  This helps detect use-after-free errors.
- *
- * Note that in the context of GC, 'use after free' is really premature garbage
- * collection.
```

Comment:
Can you implement this functionality in your allocator as well when asan is enabled?  This is useful for finding bugs.